### ISSUE TITLE
Istio blocks connection to external service

### ISSUE THREAD

**Issue by catper** [istio/istio#23332]
**Bug description**
I have two services that need to talk to each other, over https. When both services are outside the service mesh this works as expected but when pulling the services into the mesh communication no longer works, all requests time out.

`global.outboundTrafficPolicy.mode` is set to `ALLOW_ANY`

I cannot see any trace of the requests in question when running with an Istio sidecar and so have no idea what is going on or how to debug this. I've seen a number of issues around indicating that this might be a problem with `x-forwarded-proto` incorrectly being set to `http` rather than `https` but as there are no traces of any activity in the logs I have no way of verifying if this is what's happening here. I have also attempted to apply lua filters to the outbound sidecar traffic, explicitly setting `x-forwarded-proto` to `https` but that doesn't have any effect.

Any indication at all as to what is happening here would be much appreciated.

**Expected behavior**
The services can talk to each other whether inside or outside of the mesh.

**Steps to reproduce the bug**
Deploy two services and enforce tls on one of them. Have service one make a request to service two (via https) and notice how said request invariably times out. 

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
```
$ istioctl version --remote
client version: 1.5.1
control plane version: 1.5.1
data plane version: 1.5.1 (9 proxies)

$ kubectl version
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:16:51Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"14+", GitVersion:"v1.14.9-eks-502bfb", GitCommit:"502bfb383169b124d87848f89e17a04b9fc1f6f0", GitTreeState:"clean", BuildDate:"2020-02-07T01:31:02Z", GoVersion:"go1.12.12", Compiler:"gc", Platform:"linux/amd64"}

$ helm version
version.BuildInfo{Version:"v3.0.2", GitCommit:"19e47ee3283ae98139d98460de796c1be1e3975f", GitTreeState:"clean", GoVersion:"go1.13.5"}
```

**How was Istio installed?**
`istio operator init` 

Operator settings:
```
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
metadata:
  namespace: istio-system
  name: istiocontrolplane
spec:
  addonComponents:
    grafana:
      enabled: true
    kiali:
      enabled: true
  values:
    gateways:
      istio-ingressgateway:
        sds:
          enabled: true
    kiali:
      dashboard:
        viewOnlyMode: true
```

**Environment where bug was observed (cloud vendor, OS, etc)**
AWS cluster

**Comment by howardjohn** [comment]
Might be https://github.com/istio/istio/issues/16458?

**Comment by rishiraaz** [comment]
Facing a similar issue where even though ALLOW_ANY is set, i had to create serviceentries to enable outside communication. I have a lot of calls to outside and is not possible to add all as service entries

**Comment by maflaven** [comment]
I'm seeing the same issue with 1.6.1 and the following config:
```yaml
apiVersion: install.istio.io/v1alpha1
kind: IstioOperator
spec:
  values:
    global:
      proxy:
        tracer: zipkin
      controlPlaneSecurityEnabled: false
      disablePolicyChecks: true
      enableTracing: true
      mtls:
        auto: false
        enabled: false
      defaultNodeSelector:
        node_type: cpu
        spotinst.io/node-lifecycle: od
      outboundTrafficPolicy:
        mode: ALLOW_ANY
    certmanager:
      enabled: false
    gateways:
      enabled: true
      istio-ingressgateway:
        enabled: true
        serviceAnnotations: {}
      istio-egressgateway:
        enabled: false
    grafana:
      enabled: false # using separately maintained installation
    istio_cni:
      enabled: false
    kiali:
      enabled: true
      dashboard:
        auth:
          strategy: anonymous
        viewOnlyMode: true
      security:
        enabled: false
    mixer:
      policy:
        enabled: false # if enabled, global.disablePolicyChecks has affect
      telemetry:
        enabled: true
    prometheus:
      enabled: false
    tracing:
      enabled: true
      provider: jaeger
    pilot:
      traceSampling: 1.0
```

@howardjohn @mandarjog what debug info can I give you?

**Comment by howardjohn** [comment]
I am fairly certain this is the result of https://github.com/istio/istio/issues/16458 and/or https://github.com/istio/istio/pull/24763, both of which are fixed in the latest 1.6.x release. Please try this out. If oyu still see issues, post a new issue with access logs and config dump from the client - see https://github.com/istio/istio/wiki/Troubleshooting-Istio for more info. Thanks!
